internal/metamorph/stats_collector.go

```diff
@@ -127,30 +127,30 @@ func (p *Processor) StartCollectStats() error {
 		p.stats.healthyPeerConnections,
 	)
 	if err != nil {
+		p.waitGroup.Done()
 		return err
 	}
 	go func() {
 		defer func() {
 			if r := recover(); r != nil {
```

Comment:
You removed the logging stack from recover in other parts of the code. Shouldn't this one be removed too?

Reply:
After removing this some tests failed. Still having the discussion about this with Michi.